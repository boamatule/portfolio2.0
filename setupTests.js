import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';

Enzyme.configure({ adapter: new Adapter() });


// import { configure } from 'enzyme';
// import Adapter from '@cfaester/enzyme-adapter-react-18';

// configure({ adapter: new Adapter() });