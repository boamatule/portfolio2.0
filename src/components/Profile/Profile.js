/* eslint-disable react/function-component-definition */
import React from 'react';
import styled from 'styled-components';
import tw from 'twin.macro';
// import Boa from '../../images/moi.webp';

const ProfileContainer = styled.div`
	min-height: 400px;
	${tw`flex justify-between w-full pl-3 pr-3 lg:pl-12 lg:pr-12 max-w-screen-2xl`};
`;

const LeftProfileContainer = styled.div`
	${tw`flex flex-col w-1/2`};
`;

const RightProfileContainer = styled.div`
  ${tw`flex flex-col w-1/2`};
`;

const Slogan = styled.h1`
  ${tw`text-4xl font-bold text-black mb-4`};
`;

const Description = styled.p`
  ${tw`text-2xl font-bold text-black mb-4`};
`;

const Profiledescription = styled.p`
  ${tw`text-2xl font-bold text-black mb-4 w-3/4 lg:w-full`};
`;

const ProfileImageContainer = styled.div`
  ${tw`flex justify-center items-center`};
`;

const ProfileImageWrapper = styled.div`
  ${tw`flex justify-center items-center`};
`;

const ProfileImageBackground = styled.div`
  ${tw`w-full h-full bg-contain bg-center`};
`;

const ProfileImage = styled.img`
  ${tw`w-1/2 h-1/2`};
`;
const backgroundImage = "../../images/background.svg";
const Profile = () => {
  return (
    <ProfileContainer>
      <LeftProfileContainer>
        <Slogan>Hi, I'm <span style={{ color: 'purple' }}>Boa Matule</span></Slogan>
        <Description>I'm a Full Stack Developer</Description>
        <Profiledescription>I'm a Full Stack Software Developer with a diverse background spanning from the hospitality industry as a chef to E-commerce as a Co-founder and Manager. Currently, I'm based in the beautiful city of Oslo, Norway, but I'm originally from the pearl of the Indian Ocean, <span style={{ color: 'purple' }}>Mozambique </span>. I'm passionate about building, architecting, and scaling web and mobile applications using cutting-edge technology stacks. Let's create something amazing together!</Profiledescription>
      </LeftProfileContainer>
      <RightProfileContainer>
        <ProfileImageContainer>
          <ProfileImageBackground backgroundImage={backgroundImage}>
            <ProfileImage src="../../images/moi.webp" alt="Boa" />

          </ProfileImageBackground>
        </ProfileImageContainer>
      </RightProfileContainer>

    </ProfileContainer>
  );
}

export default Profile;