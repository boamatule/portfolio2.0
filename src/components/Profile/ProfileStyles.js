// import styled from 'styled-components';
// import { Container } from '../../global-styles';

// export const ProfileWrapperContainer = styled(Container)`
//   margin-bottom: 0px;
// `;
// export const ProfileWrapperRow = styled.div`
//   display: grid;
//   grid-template-columns: 1fr;
//   margin: 0px 0px auto;

//   @media screen and (max-width: 800px) {
//     justify-content: center;
//     align-items: center;
//     padding: 0px 0px auto;
//     grid-template-columns: 1fr;
//   }
// `;

// export const ProfileWrapperColumn = styled.div`
//   width: 100%;
// `;

// export const ProfileWrapper = styled.div`
//   justify-content: center;
//   align-items: center;
//   text-align: center;
//   margin-top: 120px;

//   @media screen and (max-width: 800px) {
//     justify-content: cente
//     align-items: center;
//     padding: 0px 0px auto;
//   }
// `;

// export const Heading = styled.h2`
//   font-weight: normal;
//   letter-spacing: 4px;
//   line-height: 20px;
//   justify-content: center;
//   /* justify-content: flex-start; */
//   text-align: center;
//   align-items: center;
//   font-size: 18px;
//   color: ${({ lightText }) => (lightText ? '#5454CB' : '#6969FB')};
//   padding-top: 10px;

//   @media screen and (max-width: 800px) {
//     font-size: 14px;
//     line-height: 1.8rem;
//     letter-spacing: 1.1px;
//     justify-content: center;
//     align-items: center;
//   }
// `;

// export const Avatar = styled.div`
//   position: relative;
//   margin-top: 20px;
//   align-items: center;
//   display: inline-flex;

//   @media screen and (max-width: 800px) {
//     display: flex;
//     flex-direction: column;
//     align-items: center;
//     margin-top: 20px;
//   }
// `;

// export const Img = styled.img`
//   width: 144px;
//   height: 144px;
//   border-radius: 50%;
//   margin-bottom: 20px;

//   @media screen and (min-width: 800px) {
//     width: 144px;
//     height: 144px;
//   }
// `;

// export const Subtitle = styled.p`
//   margin: 40px 20px auto;
//   color: grey;
//   font-size: 16px;
//   letter-spacing: 2px;
//   line-height: 20px;
//   font-weight: normal;
//   justify-content: flex-start;
//   text-align: center;
//   background: transparent;

//   @media screen and (max-width: 800px) {
//     font-size: 14px;
//     margin-top: 0px;
//     letter-spacing: 2px;
//     line-height: 20px;
//     font-weight: normal;
//     justify-content: center;
//     text-align: center;
//     margin: 20px 20px auto;
//   }
// `;

// export const Location = styled.p`
//   margin-top: 10px;
//   font-size: 1.1rem;
//   letter-spacing: 2px;
//   line-height: 20px;
//   color: grey;
//   font-weight: normal;
//   margin-bottom: 0px;
// `;

import styled from 'styled-components';
import { Container } from '../../global-styles';

export const ProfileWrapperContainer = styled(Container)`
  margin-bottom: 0px;
`;
export const ProfileWrapperRow = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  margin: 0px 0px auto;
  @media screen and (max-width: 800px) {
    justify-content: center;
    align-items: center;
    padding: 0px 0px auto;
    grid-template-columns: 1fr;
  }
`;

export const ProfileWrapperColumn = styled.div`
  width: 100%;
`;

export const ProfileWrapper = styled.div`
  align-items: center;
  text-align: center;
  margin-top: 120px;
  margin-bottom: 20px;
  padding: 80px 80px;
  @media screen and (max-width: 800px) {
    justify-content: flex-start;
    align-items: center;
    padding: 0px 0px;
    margin-bottom: 20px;
  }
`;

export const Heading = styled.h2`
  font-weight: normal;
  letter-spacing: 2px;
  line-height: 20px;
  justify-content: center;
  align-items: center;
  font-size: 16px;
  color: ${({ lightText }) => (lightText ? 'rgb(255, 255, 0)' : '#6969FB')};
  padding-top: 10px;
  @media screen and (max-width: 800px) {
    font-size: 14px;
    line-height: 1.8rem;
    letter-spacing: 1.1px;
    justify-content: center;
    align-items: center;
  }
`;

export const Avatar = styled.div`
  position: relative;
  margin-top: 20px;
  align-items: center;
  display: inline-flex;
  @media screen and (max-width: 800px) {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 20px;
  }
`;

export const Img = styled.img`
  width: 144px;
  height: 144px;
  border-radius: 50%;
  margin-bottom: 20px;
  @media screen and (min-width: 800px) {
    width: 144px;
    height: 144px;
  }
`;

export const Subtitle = styled.h3`
  margin-top: 10px;
  color: grey;
  font-size: 16px;
  letter-spacing: 2px;
  line-height: 20px;
  font-weight: normal;
  justify-content: flex-start;
  /* align-items: center; */
  background: transparent;
  padding: 0px 40px;

  @media screen and (max-width: 800px) {
    font-size: 14px;
    margin-top: 20px;
    justify-content: center;
  }
`;

export const Location = styled.p`
  margin-top: 10px;
  font-size: 1.1rem;
  letter-spacing: 2px;
  line-height: 20px;
  color: grey;
  font-weight: normal;
  margin-bottom: 0px;
`;
