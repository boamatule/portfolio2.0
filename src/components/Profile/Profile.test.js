import { shallow } from 'enzyme';
import Profile from './Profile';
import '../../../setupTests';

it('expect to render Profile Component', () => {
  expect(shallow(<Profile />)).toMatchSnapshot();
});
