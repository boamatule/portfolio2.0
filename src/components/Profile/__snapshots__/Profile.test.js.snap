// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Profile Component 1`] = `ShallowWrapper {}`;
